docs/docker-stack/build.rst

```diff
@@ -0,0 +1,380 @@
+    RUN apt-get update \
+      && apt-get install -y --no-install-recommends \
+             my-awesome-apt-dependency-to-add \
+      && apt-get autoremove -yqq --purge \
+      && apt-get clean \
+      && rm -rf /var/lib/apt/lists/*
+    USER airflow
+
+
+* PyPI dependencies in Apache Airflow are installed in the user library, of the "airflow" user, so
+  you need to install them with the ``--user`` flag and WITHOUT switching to airflow user. Note also
+  that using --no-cache-dir is a good idea that can help to make your image smaller.
+
+  .. code-block:: dockerfile
+
+    FROM apache/airflow:2.0.1
+    RUN pip install --no-cache-dir --user my-awesome-pip-dependency-to-add
+
+* As of 2.0.1 image the ``--user`` flag is turned on by default by setting ``PIP_USER`` environment variable
+  to ``true``. This can be disabled by un-setting the variable or by setting it to ``false``.
+
+
+* If your apt, or PyPI dependencies require some of the build-essentials, then your best choice is
+  to follow the "Customize the image" route. However it requires to checkout sources of Apache Airflow,
+  so you might still want to choose to add build essentials to your image, even if your image will
+  be significantly bigger.
+
+  .. code-block:: dockerfile
+
+    FROM apache/airflow:2.0.1
+    USER root
+    RUN apt-get update \
+      && apt-get install -y --no-install-recommends \
+             build-essential my-awesome-apt-dependency-to-add \
+      && apt-get autoremove -yqq --purge \
+      && apt-get clean \
+      && rm -rf /var/lib/apt/lists/*
+    USER airflow
+    RUN pip install --no-cache-dir --user my-awesome-pip-dependency-to-add
+
+* You can also embed your dags in the image by simply adding them with COPY directive of Airflow.
+  The DAGs in production image are in ``/opt/airflow/dags`` folder.
+
+Customizing the image
+---------------------
+
+Customizing the image is an alternative way of adding your own dependencies to the image - better
+suited to prepare optimized production images.
+
+The advantage of this method is that it produces optimized image even if you need some compile-time
+dependencies that are not needed in the final image. You need to use Airflow Sources to build such images
+from the `official distribution folder of Apache Airflow <https://downloads.apache.org/airflow/>`_ for the
+released versions, or checked out from the GitHub project if you happen to do it from git sources.
+
+The easiest way to build the image is to use ``breeze`` script, but you can also build such customized
+image by running appropriately crafted docker build in which you specify all the ``build-args``
+that you need to add to customize it. You can read about all the args and ways you can build the image
+in :doc:`build-arg-ref`.
+
+Here just a few examples are presented which should give you general understanding of what you can customize.
+
+This builds production image in version 3.6 with default extras from the local sources (master version
```

Comment:
We had these examples repeated twice in the documentation - once in the "Customizing the image" section and the second time under the tables in the "Production image build arguments" section. Some examples were common and some were different. Now we have one place with examples.